src/inmanta/agent/in_process_executor.py

```diff
@@ -68,20 +68,55 @@ def __init__(
         self.thread_pool: ThreadPoolExecutor = ThreadPoolExecutor(1, thread_name_prefix="Pool_%s" % self.name)
 
         self._cache = inmanta.agent.cache.AgentCache(self)
+        # This lock ensures cache entries can not be cleaned up when
+        # the executor is actively working and vice versa
+        self.activity_lock = asyncio.Lock()
 
         self.logger: logging.Logger = parent_logger.getChild(self.name)
 
         self._stopped = False
 
         self.failed_resource_types: FailedResourcesSet = set()
 
-    def stop(self) -> None:
+        self.cache_cleanup_tick_rate = inmanta.agent.config.agent_cache_cleanup_tick_rate.get()
+        self.periodic_cache_cleanup_job: Optional[asyncio.Task[None]] = None
+
+    async def start(self) -> None:
+        self.periodic_cache_cleanup_job = asyncio.create_task(self.cleanup_stale_cache_entries())
+
+    async def cleanup_stale_cache_entries(self) -> None:
+        """
+        Periodically cleans up stale entries in the cache. The clean_stale_entries
+        has to be called on the thread pool because it might call finalizers.
+        """
+        reschedule_interval: int = self.cache_cleanup_tick_rate
+        while not self._stopped:
+            async with self.activity_lock:
+                if self._stopped:
+                    return
+                try:
+                    await asyncio.get_running_loop().run_in_executor(self.thread_pool, self._cache.clean_stale_entries)
+                except Exception:
+                    # Make sure we don't drop out of the while loop if an exception occurs.
+                    pass
```

Comment:
Let's also log a message. Otherwise we are not aware about the fact that there is an issue.
```suggestion
                    # Make sure we don't drop out of the while loop if an exception occurs.
                    LOGGER.exception("An unexpected exception occurred while cleaning up the cache of agent %s.", self.name)
```